services/src/test/java/io/scalecube/services/ServiceUnitTest.java

```diff
@@ -0,0 +1,134 @@
+    ServiceReference reference = new ServiceReference("a", "b", Address.create("localhost", 4000));
+    ServicesConfig config = ServicesConfig.empty();
+
+    ServiceProcessor processor = new AnnotationServiceProcessor();
+    ServiceRegistry registry = new ServiceRegistryImpl(member.cluster(), config, processor);
+    RemoteServiceInstance instance = new RemoteServiceInstance(registry, reference, new HashMap<>());
+
+    assertTrue(instance.toString().equals("RemoteServiceInstance [address=localhost:4000, memberId=a]"));
+    assertTrue(instance.tags().isEmpty());
+    assertTrue(instance.memberId().equals("a"));
+    assertTrue(instance.address().equals(Address.create("localhost", 4000)));
+    assertTrue(!instance.isReachable());
+    assertTrue(!instance.isLocal());
+    assertTrue(instance.serviceName().equals("b"));
+  }
+
+  @Test
+  public void test_service_reference() {
+    Address addr = Address.create("localhost", 4000);
+    ServiceReference reference = new ServiceReference("a", "b", addr);
+    assertTrue(reference.address().equals(addr));
+    assertTrue(reference.memberId().equals("a"));
+    assertTrue(reference.serviceName().equals("b"));
+
+    ServiceReference aref = new ServiceReference("a", "b", addr);
+    ServiceReference bref = new ServiceReference("a", "b", addr);
+
+    assertTrue(aref.equals(aref));
+    assertTrue(aref.equals(bref));
+    assertTrue(!aref.equals(null));
+
+    assertTrue(reference.toString().equals("ServiceReference [memberId=a, qualifier=b, address=localhost:4000]"));
+  }
+
+  @Test
+  public void test_service_config() {
+
+    Microservices.Builder builder = Microservices.builder();
+    ServicesConfig config =
+        ServicesConfig.builder(builder).service(new GreetingServiceImpl()).tag("a", "b").add().create();
+    builder.build();
+
+    assertTrue(!config.getServiceConfigs().isEmpty());
+    assertTrue(!config.getServiceConfigs().get(0).getDefinitions().isEmpty());
+
+
+  }
+
+  @Test
+  public void test_router_factory() {
+    RouterFactory factory = new RouterFactory(null);
+    Router router = factory.getRouter(RandomServiceRouter.class);
+    assertTrue(router != null);
+    
+    Router dummy = factory.getRouter(DummyRouter.class);
+    assertTrue(dummy == null);
+    
+  }
+
+  @Test
+  public void test_dispatching_future()
+      throws IllegalArgumentException, IllegalAccessException, NoSuchFieldException, SecurityException {
+    Microservices member = Microservices.builder().build();
+    ServiceResponse response = new ServiceResponse(fn -> {
+      return null;
+    });
+
+    Message request = Message.builder().correlationId(response.correlationId()).build();
+    Field field = Message.class.getDeclaredField("sender");
+    field.setAccessible(true);
+    field.set(request, member.cluster().address());
+
+    DispatchingFuture dispatcher = DispatchingFuture.from(member.cluster(), request);
+    dispatcher.complete(new Throwable());
+
+    assertTrue(true);
```

Comment:
yes, but since the only assert you have here is `assertTrue(true)` which obviously passing no matter what you will do above it is kind of allude that it didn't really test anything.